packages/web/src/views/governance/components/proposals-list/view-proposal-modal/VoteCtaButton.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useMemo } from "react";
+import { useTranslation } from "react-i18next";
+
+import Button, { ButtonHierarchy } from "@components/common/button/Button";
+import { DEVICE_TYPE } from "@styles/media";
+
+interface VoteCtaButtonProps {
+  breakpoint: DEVICE_TYPE;
+  isWalletConnected: boolean;
+  isSwitchNetwork: boolean;
+  voteType?: string;
+  voteWeigth?: number;
+  status: string;
+  selectedVote: string;
+  handleVote: (voteType: boolean) => void;
+  connectWallet: () => void;
+  switchNetwork: () => void;
+}
+
+const VoteCtaButton: React.FC<VoteCtaButtonProps> = ({
+  breakpoint,
+  isWalletConnected,
+  isSwitchNetwork,
+  voteType,
+  voteWeigth,
+  status,
+  selectedVote,
+  handleVote,
+  connectWallet,
+  switchNetwork,
+}) => {
+  const { t } = useTranslation();
+
+  const btnStatus: {
+    disabled: boolean;
+    text: string;
+    action: (() => void) | null;
+  } = useMemo(() => {
+    if (status === "UPCOMMING")
+      return {
+        disabled: true,
+        text: t("Governance:proposal.status.upcomming"),
+        action: null,
+      };
+    if (status === "REJECTED")
+      return {
+        disabled: true,
+        text: t("Governance:proposal.status.rejected"),
+        action: null,
+      };
+    if (status === "CANCELLED")
+      return {
+        disabled: true,
+        text: t("Governance:proposal.status.cancelled"),
+        action: null,
+      };
+    if (status === "EXECUTED")
+      return {
+        disabled: true,
+        text: t("Governance:proposal.status.executed"),
+        action: null,
+      };
+    if (!isWalletConnected)
+      return {
+        disabled: false,
+        text: t("Governance:detailModal.btn.login"),
+        action: connectWallet,
+      };
+    if (isSwitchNetwork)
+      return {
+        disabled: false,
+        text: t("Governance:detailModal.btn.switchNetwork"),
+        action: switchNetwork,
+      };
```

Comment:
Wouldn't it be fine if this was managed as a state?
Text key values already exist. (`Governance:detailModal.btn.login`, `Governance:detailModal.btn.switchNetwork `)

Reply:
it could be confused.
`status` in `VoteCtaButton`'s prop means proposal status, 
and this statement set btnStatus inside the component